boards/arm/nucleo_f042k6/nucleo_f042k6.yaml

```diff
@@ -0,0 +1,18 @@
+identifier: nucleo_f042k6
+name: ST Nucleo F042K6
+type: mcu
+arch: arm
+toolchain:
+  - zephyr
+  - gnuarmemb
+  - xtools
+ram: 6
```

Comment:
On some other devices with limited available RAM, gpio d and f where disabled.
Is it safe to keep them?
Did you run compilation campaign on this board to detect test with potential RAM overflow ?